prereise/gather/demanddata/uci/HiGrid_Elec_Trans4_8_nths.py

```diff
@@ -0,0 +1,195 @@
+    '''
+    # keep track of which rows to delete
+    rows_to_delete = set()
+    for vtype in range(len(newdata)):
+        # >4 for LDV, <5 for LDT
+        if newdata[vtype][16] < 5:
+            rows_to_delete.add(vtype)
+
+    # create a new list for all the data values that don't include the rows we want to remove
+    data = []
+    for i in range(len(newdata)):
+        if i not in rows_to_delete:
+            data.append(newdata[i])
+
+    return data
+
+
+#ldt = light duty truck
+#ldv = light duty vehicle
+def total_daily_vmt(census_1: int, Comm_type: int, locationstrategy: list[int], inputday: list[int]) -> np.array[list[float]]
+    '''loads data and uses the parameters to calculate total_daily_vmt
+
+    :param int census_1: the type of census
+    :param int Comm_type: the type of Commute
+    :param list[int] locationstrategy: strategy for each location
+    :param list[int] inputday: day of the week for each day in the year derived from first_func
+    :return: list[[int, int]] -- daily_vmt_total each row is a year of entries for each vehicle type
+    '''
+    # get the data
+    data = np.array(remove_ldt(load_data(census_1)))
+    n = len(data)
+    # removes VMT for trips from work->home and home-> work
+    # (not exactly correct due to chained trips involving work and home but no
+    # way to remove the data)
+    if Comm_type == 1:
+        if all([i != 2 for i in locationstrategy]):
+            locationstrategy = 2                    #putting this part in function means rest of code won't be able to see change in locationstrategy value
+                                                    #will have to use global or return locationstrategy and reassign
+            print('"locationstrategy" changed to "Home and Work" for Comm_type == 1')
+        # isolates home->work trips
+        trip_home_work = [i for i in range(len(data)) if (data[i][14] == 1 + data[i][15]== 11) == 2]
+        # isolates work->home trips
+        trip_work_home = [i for i in range(len(data)) if (data[i][14] == 11 + data[i][15]== 1) == 2]
+
+        # set trips from work to home and vice versa to 0 in column 12 (0 indexed, 13 1 indexed)
+        for trip in trip_home_work:
+            data[trip][12] = 0
+
+        for trip in trip_work_home:
+            data[trip][12] = 0
+
+    # pre-processing of vehicle trip data to determine VMT and trips per day to speed up later operation
+    month2 =  get_month2(data)
+    dayofweek2 = get_dayofweek2(data)
+
+    for its in range(len(data)): #kef
+        # weekend in the case of 1 or 7
+        if data[its][6] == 1 or data[its][6] == 7:
+            data[its][7] = 1
+        # weekday in the case of 2-6 (inclusive)
+        elif 2 <= data[its][6] and data[its][6] <= 6:
+            data[its][7] = 2
+
+    day2 = get_day2(data)
+    daily_vmt_total = np.array([[0, 0] for i in range(365)])
+
+    for day_iter in range(365):
+        for i in range(n):
+            if day2[i] == inputday[day_iter]:

```

Comment:
If `day2` is a vector of weekday-or-weekend, then why do we compare this with the day of the week in `inputday`?